packages/xo-web/src/common/xo-compression/select-compression.js

```diff
@@ -0,0 +1,53 @@
+import _ from 'intl'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { injectState, provideState } from 'reaclette'
+import { omit } from 'lodash'
+
+import decorate from '../apply-decorators'
+import Select from '../form/select'
+
+const COMMON_OPTIONS = [
+  {
+    label: _('stateDisabled'),
+    value: '',
+  },
+  {
+    label: _('chooseCompressionGzipOption'),
+    value: 'native',
+  },
+]
+
+const ZSTD_OPTION = {
+  label: _('chooseCompressionZstdOption'),
+  value: 'zstd',
+}
+
+const SELECT_COMPRESSION_PROP_TYPES = {
+  showZstd: PropTypes.bool,
+}
+
+const SelectCompression = decorate([
+  provideState({
+    computed: {
+      options: (_, { showZstd }) =>
+        showZstd ? [...COMMON_OPTIONS, ZSTD_OPTION] : COMMON_OPTIONS,
+      selectProps: (_, props) =>
+        omit(props, Object.keys(SELECT_COMPRESSION_PROP_TYPES)),
```

Comment:
`omit(props, 'zstd')`  ? 
I still see,  using `Object.keys` is complex.
@pdonias  what do you think?

Reply:
I think it's fine since it's a very small object anyways. It could be optimized by computing the keys only once but it's negligible.